tor_archivist/main.py

```diff
@@ -63,11 +83,19 @@ def run(config):
         # me_irl explosion
         if flair == css_flair.completed:
             logging.info(f'Archiving completed post "{post.title}"...')
-            config.archive.submit(
-                post.title,
-                url=reddit_url.format(post.permalink))
+
+            # look for the transcription
+            transcript = find_transcription(original_post)
+
+            if transcript is not None:
+                config.archive.submit(
+                    post.title,
+                    url=reddit_url.format(transcript.permalink))
+                logging.info('Post archived!')
+            else:
+                logging.info('Could not find the transcript - won\'t archive.')
```

Comment:
We might want this to be a warn — it's a bug if we can't find the transcript on a post. Also log the URLs, and other relevant information.

Reply:
I suppose in some cases it should be a warning, but 99% of the time this will happen, it's because a transcript was autoremoved, in which case we just treat it like completed (see @itsthejoker's history validation PR on ToR). We already have that as a sort of warning by sending it to a separate channel in our slack. If there were an easy way to detect the other cases, that is, when something has actually gone wrong on our end, I'd definitely be in favour of making that an error/warning, but right now there isn't, and it's not really the most crucial part of the platform anyway.

Reply:
Ohhh that's right. That's gonna stink.

It'd be neat if we added history validation, and transcription validation in general to tor_core, but that's not relevant to this PR.

Reply:
I'm also curious why we're avoiding archiving if it can't find the transcript. I know there's the spam filter and that it might remove a completed transcription comment, but would we want it to stick around in /r/ToR's feed afterwards?

I suppose this also lends to the idea that we might want to link to the detected transcription comment in /r/ToR post to get around the deletion thing. 🤷‍♂️